### PR TITLE
Fixed the issue where edit textBox loses focus when using up/down to switch property values ​​on edit textBox

### LINKED ISSUE #12440: When using up/down to toggle property value on edit text box, edit text box loses focus
### .NET version

10.0.100-alpha.1.24553.6

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, all versions have this issue

### Issue description

The rectangle always back to focus on entire row after using the up/down changed the `AutoSize ` property values directly in the edit textbox

BeforeChanges

**Expected result:**
The rectangle in the edit text box always stays in the edit textbox when changing the property value using the up/down buttons 
AfterChanges

### Steps to reproduce

1. Create a Windows Forms App
2. Add a Button to form and focus on it
3. Open the property page
4. Open Inspect
5. Select `AutoSize` property
6. Changing the property value using the up/down buttons in the edit textbox

### PR DESCRIPTION
**Pull request by LeafShi1**
Fixes #12440

## Proposed changes

- When setting a property with the [RefreshProperties(RefreshProperties.All)] attribute, suppress recalculation and reset constants for the entire property page

## Customer Impact

- The rectangle in the edit text box always stays in the edit textbox when changing the property value using the up/down buttons 

## Regression? 

- No

## Risk

- Mininal

### Before
The rectangle always back to focus on entire row after using the up/down changed the property values directly in the edit textbox
BeforeChanges

### After
The rectangle in the edit text box always stays in the edit textbox when changing the property value using the up/down buttons 
AfterChanges

## Test methodology 

- Manually

## Test environment(s) 

- .net 10.0.0-alpha.1.24554.9

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12431)

### CODE REVIEW

**Comment by LeafShi1** [comment]
The private dlls have been tested, no regression issue was found.

**Comment by Tanya-Solyanik** [comment]
@LeafShi1  - please fix a link to the issue, right now it points to another PR

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/PropertyGridView.cs [hunk 1]
```diff
@@ -3997,7 +3997,7 @@ private void Refresh(bool fullRefresh, int startRow, int endRow)
             startRow = 0;
         }
 
-        if (fullRefresh || OwnerGrid.HavePropertyEntriesChanged())
+        if (OwnerGrid.HavePropertyEntriesChanged())
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/PropertyGridView.cs:4000]: COuld you please help me understand this fix? I would expect that in this scenario OwnerGrid.HavePropertyEntriesChanged is true.

Comment (LeafShi1) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/PropertyGridView.cs:4000]: When setting a property, the SetPropertyValue method is called. In this method, the Refresh method is called to refresh the property list based on the Attribute of the property.

But in fact, when setting the AutoSize property, the number and structure of properties in the property list will not change. We only need to refresh the linked property values, so there is no need to recreate the property list.

Comment (LeafShi1) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/PropertyGridView.cs:4000]: When we switch the value of the edit text by pressing the up and down keys, the `SelectAll` event will be called, and this event will eventually enter the `SelectInternal` method. The AI ​​notification is sent from the `SelectInternal` method here, but the notification sent here is `UIA_Text_TextSelectionChangedEventId`. If we add notification `UIA_AutomationFocusChangedEventId` to it,  this problem also can be resolved.

image

**Review by ricardobossan — COMMENTED** [review]
Tested the changes and the TextBox does not lose focus when you press up/down.

Changes LGTM.

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/PropertyGridView.cs (hunk 1 above)
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/PropertyGridView.cs]: >But in fact, when setting the AutoSize property, the number and structure of properties in the property list will not change. We only need to refresh the linked property values, so there is no need to recreate the property list.

Please verify this with a Label control and Multiline property

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/PropertyGridView.cs]: Please test with NVDA and JAWS.

**Review by Tanya-Solyanik — APPROVED** [review]

**Comment by Olina-Zhang** [comment]
Tested this PR change with several regression issues found. Need to investigate.

**Comment by LeafShi1** [comment]
The second commit caused 4 regression issues, `RaiseAutomationEvent(UIA_EVENT_ID.UIA_AutomationFocusChangedEventId)` does not stably focus the accessibility rectangle on the edit text box, so consider the solution **When setting a property with the [RefreshProperties(RefreshProperties.All)] attribute, suppress recalculation and reset constants for the entire property page** in commit1. And commit1 has been tested before, no regression issue.

**Review by Tanya-Solyanik — APPROVED** [review]
ok, let's remove the `fullRefresh` test.

**Pull request merged**